packages/docregistry/src/registry.ts

```diff
@@ -401,34 +393,34 @@ export class DocumentRegistry implements IDisposable {
     }
 
     // Add the file type factories in registration order.
-    fts.forEach(ft => {
+    for (const ft of fts) {
       if (ft.name in this._widgetFactoriesForFileType) {
-        each(this._widgetFactoriesForFileType[ft.name], n => {
+        for (const n of this._widgetFactoriesForFileType[ft.name]) {
           factories.add(n);
-        });
+        }
       }
-    });
+    }
 
     // Add the rest of the global factories, in registration order.
     if ('*' in this._widgetFactoriesForFileType) {
-      each(this._widgetFactoriesForFileType['*'], n => {
+      for (const n of this._widgetFactoriesForFileType['*']) {
         factories.add(n);
-      });
+      }
     }
 
     // Construct the return list, checking to make sure the corresponding
     // model factories are registered.
     const factoryList: DocumentRegistry.WidgetFactory[] = [];
-    factories.forEach(name => {
+    for (const name of factories) {
       const factory = this._widgetFactories[name];
       if (!factory) {
-        return;
+        break;
```

Comment:
I think we need to introspect all possible factories.

```suggestion
        continue;
```